primo/data_parser/input_options_well_data.py

```diff
@@ -0,0 +1,214 @@
+# perform publicly and display publicly, and to permit others to do so.
+#################################################################################
+
+"""
+This module defines all the input options for objects of WellData class
+"""
+
+
+# Installed libs
+from pyomo.common.config import (
+    Bool,
+    ConfigDict,
+    ConfigValue,
+    In,
+    NonNegativeFloat,
+    NonNegativeInt,
+)
+
+# User libs
+from primo.utils.domain_validators import InRange
+from primo.data_parser.default_data import SUPP_IMPACT_METRICS
+
+
+# This dictionary contains default values to fill with,
+# if the data is not provided.
+binary_type_metrics = {
+    "leak": True,  # Well is leaking
+    "compliance": True,  # Well is compliant
+    "violation": False,  # Well is not in violation
+    "incident": False,  # Incident is assumed to be False
+    "h2s_leak": False,  # H2S Leak is absent
+    "brine_leak": False,  # Brine leak is absent
+    "buildings_near": False,  # No buildings nearby
+    "buildings_far": False,  # No distant buildings
+    "fed_wetlands_near": False,  # No federal wetlands nearby
+    "fed_wetlands_far": False,  # No distant federal wetlands
+    "state_wetlands_near": False,  # No state wetlands nearby
+    "state_wetlands_far": False,  # No distant state wetlands
+    # Assuming that well integrity is good, if it is not provided
+    "well_integrity": False,
+}
+
+# Input options for well data class
+CONFIG = ConfigDict()
+CONFIG.declare(
+    "census_year",
+    ConfigValue(
+        default=2020,
+        domain=In(list(range(2020, 2101, 10))),
+        doc="Year for collecting census data",
+    ),
+)
+CONFIG.declare(
+    "ignore_operator_name",
+    ConfigValue(
+        default=True,
+        domain=Bool,
+        doc="Remove well if operator name is not provided",
+    ),
+)
+CONFIG.declare(
+    "missing_age",
+    ConfigValue(
+        default="fill",
+        domain=In(["fill", "estimate", "remove"]),
+        doc="Method for processing missing age information",
+    ),
+)
+CONFIG.declare(
+    "missing_depth",
+    ConfigValue(
+        default="fill",
+        domain=In(["fill", "estimate", "remove"]),
+        doc="Method for processing missing depth information",
+    ),
+)
+CONFIG.declare(
+    "fill_age",
+    ConfigValue(
+        default=100,
+        # Assuming that no well is older than 250 years
+        domain=InRange(0, 250),
+        doc="Value to fill with, if the age is missing",
+    ),
+)
+CONFIG.declare(
+    "fill_depth",
+    ConfigValue(
+        default=1000,
+        # Assuming that no well is deeper than 20,000 ft
+        domain=InRange(0, 20000),
+        doc="Value to fill with, if the depth is missing",
+    ),
+)
+
+for key, val in binary_type_metrics.items():
+    CONFIG.declare(
+        "fill_" + key,
+        ConfigValue(
+            default=val,
+            domain=Bool,
+            doc=(
+                f"Value to fill with, if {SUPP_IMPACT_METRICS[key].full_name} "
```

Comment:
Should we wrap the whole thing in a function instead? I can imagine a scenario where someone wants to use an extended/modified SUPP_IMPACT_METRICS dictionary and would rather not have this hardcoded in that scenario: it can be an input (which by default can still be set to SUPP_IMPACT_METRICS!

Reply:
I'm not using any impact metrics related data. I just need the full name of the metric for the doc string. Instead of duplicating the full name, I figured I can use the one in `SUPP_IMPACT_METRICS`. If this part of the code is confusing, then I will manually add the full name in the `binary_type_metrics` dict.

Reply:
It is definitely confusing. And also adds an additional layer where confusion is likely. Are the separate classes truly necessary? If yes, what if the two classes  (`binary_type_metrics` and `SUPP_IMPACT_METRICS`) are not consistent? if you are adding column names to binary_type_metrics, then are they needed in `SUPP_IMPACT_METRICS`? I am missing some context: happy to set up a call if that will be useful